### PR TITLE
Publish blockv2 ssz

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Fixes check on HTTP method type for HTTP methods and also implements ssz responses for the publishblockv2 requests

### CODE REVIEW

File: network/reader.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// match a number with optional decimals
+var priorityRegex = regexp.MustCompile(`q=(\d+(?:\.\d+)?)`)
+
+// SszRequested takes a http request and checks to see if it should be requesting a ssz response.
+func SszRequested(req *http.Request) (bool, error) {
```
Comment (james-prysm) [inline network/reader.go:14]: This code already exists in deneb-integration but needed here

**Review by terencechain — APPROVED** [review]

File: beacon-chain/rpc/eth/beacon/handlers.go [hunk 2]
```diff
@@ -38,7 +41,159 @@ func (bs *Server) PublishBlindedBlockV2(w http.ResponseWriter, r *http.Request)
...
+			Message: "Could not read request body",
+			Code:    http.StatusInternalServerError,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	capellaBlock := &ethpbv2.SignedBlindedBeaconBlockCapella{}
+	if err := capellaBlock.UnmarshalSSZ(body); err == nil {
+		if err = validate.Struct(capellaBlock); err == nil {
+			v1block, err := migration.BlindedCapellaToV1Alpha1SignedBlock(capellaBlock)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/handlers.go:74]: You can use the same `ToGeneric()` pattern as in `publishBlindedBlockV2`, no?

File: beacon-chain/rpc/eth/beacon/handlers.go [hunk 3]
```diff
@@ -178,7 +349,140 @@ func (bs *Server) PublishBlockV2(w http.ResponseWriter, r *http.Request) {
 		network.WriteError(w, errJson)
 		return
 	}
+	capellaBlock := &ethpbv2.SignedBeaconBlockCapella{}
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/handlers.go:352]: Same comments about ignoring fields and using `ToGeneric()`.

File: beacon-chain/rpc/eth/beacon/handlers.go [hunk 4]
```diff
@@ -38,7 +41,159 @@ func (bs *Server) PublishBlindedBlockV2(w http.ResponseWriter, r *http.Request)
 	if ok := bs.checkSync(r.Context(), w); !ok {
 		return
 	}
+	isSSZ, err := network.SszRequested(r)
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
```
[the remark this answers is not in the document]
Comment (james-prysm) [reply beacon-chain/rpc/eth/beacon/handlers.go:46]: yeah that works

File: beacon-chain/rpc/eth/beacon/handlers.go (hunk 2 above)
Comment (james-prysm) [reply beacon-chain/rpc/eth/beacon/handlers.go:74]: it's not the same type, this is the v2 type here to call unmarshalssz

File: beacon-chain/rpc/eth/beacon/handlers.go [hunk 5]
```diff
@@ -38,7 +41,159 @@ func (bs *Server) PublishBlindedBlockV2(w http.ResponseWriter, r *http.Request)
...
+			Message: "Could not read request body",
+			Code:    http.StatusInternalServerError,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	capellaBlock := &ethpbv2.SignedBlindedBeaconBlockCapella{}
+	if err := capellaBlock.UnmarshalSSZ(body); err == nil {
```
[the remark this answers is not in the document]
Comment (james-prysm) [reply beacon-chain/rpc/eth/beacon/handlers.go:72]: will try more tests, UnmarshalSSZ should be able to catch it, in fact the validator struct might not be needed.

**Review by rkapka — APPROVED** [review]

**Pull request merged**